platform/solana/transaction.go

```diff
@@ -2,43 +2,66 @@ package solana
 
 import (
 	"errors"
+	"fmt"
 	"strconv"
 
+	"github.com/trustwallet/blockatlas/pkg/blockatlas"
 	"github.com/trustwallet/golibs/types"
 )
 
+const (
+	programSystem = "system"
+	programToken  = "spl-token"
+
+	instructionTransfer = "transfer"
+	// will support instructionTransferChecked later
+	// instructionTransferChecked = "transferChecked"
+)
+
 func (p *Platform) GetTxsByAddress(address string) (types.TxPage, error) {
 	results := make(types.TxPage, 0)
-	txs, err := p.client.GetTransactions(address)
+	txs, err := p.client.GetTransactionsByAddress(address)
 	if err != nil {
 		return results, err
 	}
 	for _, tx := range txs {
-		if normalized, err := p.NormalizeTx(tx, address); err == nil {
+		timestamp := tx.BlockTime
+		if timestamp == 0 {
+			timestamp = EstimateTimestamp(tx.Slot)
```

Comment:
can we remove `EstimateTimestamp` ? it's not accurate

Reply:
done